arch/power/compare256_power9.c

```diff
@@ -0,0 +1,58 @@
+/* compare256_power9.c - Power9 version of compare256
+ * Copyright (C) 2019 Matheus Castanho <msc@linux.ibm.com>, IBM
+ * For conditions of distribution and use, see copyright notice in zlib.h
+ */
+
+#ifdef POWER9
+#include <altivec.h>
+#include "../../zbuild.h"
+#include "../../zendian.h"
+
+static inline uint32_t compare256_power9_static(const uint8_t *src0, const uint8_t *src1) {
+    vector unsigned char vsrc0, vsrc1, vc;
+    uint32_t len = 0, cmplen;
+
+    do {
+        vsrc0  = *((vector unsigned char *)src0);
+        vsrc1 = *((vector unsigned char *)src1);
+
+        /* Compare 16 bytes at a time. Each byte of vc will be either
+        * all ones or all zeroes, depending on the result of the comparison. */
+        vc = (vector unsigned char)vec_cmpne(vsrc0, vsrc1);
+
+        /* Since the index of matching bytes will contain only zeroes
+         * on vc (since we used cmpne), counting the number of consecutive
+         * bytes where LSB == 0 is the same as counting the length of the match. */
+#if BYTE_ORDER == LITTLE_ENDIAN
+        cmplen = __builtin_vec_vctzlsbb(vc);
+        //asm volatile("vctzlsbb %0, %1\n\t" : "=r" (len) : "v" (vc));
+#else
+        cmplen = __builtin_vec_vclzlsbb(vc);
+        //asm volatile("vclzlsbb %0, %1\n\t" : "=r" (len) : "v" (vc));
```

Comment:
We don't need the asm version as comment, we prefer intrinsics when available.

Reply:
Fixed.